File: Scripts/results/generated_code_results/baseline_gpt/Education/EnglishLearningAppUI/ConversationScreen.js
import React from 'react';
import { View, Text, StyleSheet, Image, ScrollView, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

const conversations = [
  {
    id: 1,
    user: 'Ellen',
    message: 'Hey there',
    translation: 'Xin chào',
    avatar: '../assets/snack-icon.png',
  },
  {
    id: 2,
    user: 'Taxi Driver',
    message: 'Hey, where are you going today?',
    translation: 'Chào cô, hôm nay cô sẽ đi đâu',
    avatar: '../assets/snack-icon.png',
  },
  {
    id: 3,
    user: 'Ellen',
    message: '25 Gerhold Valley, please.',
    translation: 'Cho tôi tới 25 Gerhold Valley.',
    avatar: '../assets/snack-icon.png',
  },
  {
    id: 4,
    user: 'Taxi Driver',
    message: 'No problem.',
    translation: 'Không vấn đề.',
    avatar: '../assets/snack-icon.png',
  },
];

const ConversationScreen = () => {
  return (
    <ScrollView style={styles.container}>
      {conversations.map((conversation) => (
        <View key={conversation.id} style={styles.conversationBox}>
          <Text style={styles.messageText}>{conversation.message}</Text>
          <Text style={styles.translationText}>{conversation.translation}</Text>
          <View style={styles.row}>
            <Image source={{ uri: conversation.avatar }} style={styles.avatar} />
            <Text style={styles.userText}>{conversation.user}</Text>
            <TouchableOpacity style={styles.iconButton}>
              <Icon name="volume-high-outline" size={24} color="#333" />
            </TouchableOpacity>
          </View>
          <TouchableOpacity style={styles.repeatButton}>
            <Icon name="mic-outline" size={16} color="#fff" />
            <Text style={styles.buttonText}>REPEAT</Text>
          </TouchableOpacity>
        </View>
      ))}
      <View style={styles.actionButtons}>
        <TouchableOpacity style={styles.listenButton}>
          <Text style={styles.buttonText}>LISTEN</Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.speakButton}>
          <Text style={styles.buttonText}>SPEAKING</Text>
        </TouchableOpacity>
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    paddingVertical: 10,
  },
  conversationBox: {
    backgroundColor: '#f8f9fa',
    margin: 10,
    borderRadius: 10,
    padding: 15,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 5,
    elevation: 3,
  },
  messageText: {
    fontSize: 16,
    fontWeight: 'bold',
    marginBottom: 8,
  },
  translationText: {
    fontSize: 14,
    color: '#666',
    marginBottom: 12,
  },
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  avatar: {
    width: 30,
    height: 30,
    borderRadius: 15,
  },
  userText: {
    fontSize: 14,
    color: '#333',
    flex: 1,
    marginLeft: 10,
  },
  iconButton: {
    padding: 5,
  },
  repeatButton: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#0058a3',
    borderRadius: 20,
    paddingVertical: 8,
    marginTop: 15,
  },
  buttonText: {
    color: '#fff',
    fontSize: 14,
    marginLeft: 5,
  },
  actionButtons: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginVertical: 20,
  },
  listenButton: {
    backgroundColor: '#34c759',
    paddingVertical: 10,
    paddingHorizontal: 30,
    borderRadius: 20,
    alignItems: 'center',
  },
  speakButton: {
    backgroundColor: '#007aff',
    paddingVertical: 10,
    paddingHorizontal: 30,
    borderRadius: 20,
    alignItems: 'center',
  },
});

export default ConversationScreen;

File: Scripts/results/generated_code_results/Flutter/Education/EnglishLearningAppUI/ConversationScreen.dart
import 'package:flutter/material.dart';
class ConversationScreen extends StatefulWidget {
const ConversationScreen({Key? key}) : super(key: key);
@override
_ConversationScreenState createState() => _ConversationScreenState();
}
class _ConversationScreenState extends State<ConversationScreen> {
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
leading: IconButton(
icon: Icon(Icons.arrow_back),
onPressed: () => Navigator.pop(context),
),
title: Text('When taking a taxi'),
),
body: Column(
children: [
Expanded(
child: ListView(
padding: EdgeInsets.all(16),
children: [
_buildConversationItem(
'Ellen',
'Hey there',
'Xin chào',
true,
),
_buildConversationItem(
'Taxi Driver',
'Hey,where are you going today?',
'Chào cô, hôm nay cô sẽ đi đâu',
false,
),
_buildConversationItem(
'Ellen',
'25 Gerhold Valley, please.',
'Cho tôi tới 25 Gerhold Valley.',
true,
),
_buildConversationItem(
'Taxi Driver',
'No problem.',
'Không vấn đề.',
false,
),
],
),
),
Padding(
padding: const EdgeInsets.all(16.0),
child: Row(
mainAxisAlignment: MainAxisAlignment.spaceEvenly,
children: [
ElevatedButton.icon(
icon: Icon(Icons.volume_up),
label: Text('Listen'),
onPressed: () {},
style: ElevatedButton.styleFrom(
backgroundColor: Colors.green,
),
),
ElevatedButton.icon(
icon: Icon(Icons.mic),
label: Text('Speaking'),
onPressed: () {},
style: ElevatedButton.styleFrom(
backgroundColor: Colors.blue,
),
),
],
),
),
],
),
);
}
Widget _buildConversationItem(String speaker, String text, String translation, bool isUser) {
return Padding(
padding: const EdgeInsets.symmetric(vertical: 8.0),
child: Row(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
if (!isUser) ...[
CircleAvatar(
backgroundImage: AssetImage('assets/images/template.png'),
radius: 20,
),
SizedBox(width: 8),
],
Expanded(
child: Column(
crossAxisAlignment: isUser ? CrossAxisAlignment.end : CrossAxisAlignment.start,
children: [
Text(
speaker,
style: TextStyle(fontWeight: FontWeight.bold),
),
SizedBox(height: 4),
Container(
padding: EdgeInsets.all(8),
decoration: BoxDecoration(
color: isUser ? Colors.blue[100] : Colors.grey[200],
borderRadius: BorderRadius.circular(12),
),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(text),
SizedBox(height: 4),
Text(
translation,
style: TextStyle(fontStyle: FontStyle.italic, color: Colors.grey[600]),
),
],
),
),
SizedBox(height: 4),
ElevatedButton.icon(
icon: Icon(Icons.repeat, size: 18),
label: Text('REPEAT'),
onPressed: () {},
style: ElevatedButton.styleFrom(
padding: EdgeInsets.symmetric(horizontal: 8, vertical: 4),
backgroundColor: Colors.grey[300],
foregroundColor: Colors.black,
),
),
],
),
),
if (isUser) ...[
SizedBox(width: 8),
CircleAvatar(
backgroundImage: AssetImage('assets/images/template.png'),
radius: 20,
),
],
],
),
);
}
}